CMake C compiler ABI detection fails with GCC

**Problem description**

Trying to get cmake to use `gcc` rather than `clang` fails with `Platform/Android/abi--GNU can't be found`, due to empty `${CMAKE_ANDROID_ARCH_ABI}`.

**Steps to reproduce**

* Install cmake from termux repo.
* Install gcc from its-pointless repo.
* cmake always seems to prefer clang by default.
* Change `usr/bin/gcc` to link to `gcc-10` rather than `clang-12`.
* Run the `setup-parchforgcc` script from its-pontless.
* Simple `hello-world` project dir, with `src/hello-world.c`  and `src/CMakeLists.txt` files.
* Call cmake with either `CC=gcc ...` or `... -DCMAKE_C_COMPILER=/full/path/to/gcc`, same fail result.
* Error in `Android-GNU.cmake` saying `Platform/Android/abi--GNU` can't be found.
* `${CMAKE_ANDROID_ARCH_ABI}` empty when trying to specify `gcc`.
* Using `clang` with either method builds the build system.

Not sure of root cause at this point.  Seeking clarification.

**Expected behavior**

Expect to be able to freely switch between gcc, clang, or any other compiler.

**Additional information**

Create project:
```
mkdir -p hello-world/{src,bin}
cd hello-world
echo -e '#include <stdio.h>

int main(void) {
\tprintf("Hello, World!\\n");
\treturn 0;
}' > src/hello-world.c
echo 'cmake_minimum_required(VERSION 3.20)
project(hello-world LANGUAGES C)
add_executable(hello-world hello-world.c)' > src/CMakeLists.txt
```

Build:
```
rm -rf bin/*
cmake -DCMAKE_C_COMPILER=`which gcc` -S src -B bin -DCMAKE_BUILD_TYPE=Release
```

Output:
```
-- The C compiler identification is GNU 10.2.0
CMake Error at /data/data/com.termux/files/usr/share/cmake-3.20/Modules/Platform/Android-GNU.cmake:29 (include):
  include could not find requested file:

    Platform/Android/abi--GNU
Call Stack (most recent call first):
  /data/data/com.termux/files/usr/share/cmake-3.20/Modules/Platform/Android-GNU-C.cmake:1 (include)
  /data/data/com.termux/files/usr/share/cmake-3.20/Modules/CMakeCInformation.cmake:48 (include)
  CMakeLists.txt:2 (project)


-- Detecting C compiler ABI info
CMake Error at /data/data/com.termux/files/usr/share/cmake-3.20/Modules/Platform/Android-GNU.cmake:29 (include):
  include could not find requested file:

    Platform/Android/abi--GNU
Call Stack (most recent call first):
  /data/data/com.termux/files/usr/share/cmake-3.20/Modules/Platform/Android-GNU-C.cmake:1 (include)
  /data/data/com.termux/files/usr/share/cmake-3.20/Modules/CMakeCInformation.cmake:48 (include)
  /data/data/com.termux/files/home/hello-world/bin/CMakeFiles/CMakeTmp/CMakeLists.txt:2 (project)


CMake Error at /data/data/com.termux/files/usr/share/cmake-3.20/Modules/CMakeDetermineCompilerABI.cmake:49 (try_compile):
  Failed to configure test project build system.
Call Stack (most recent call first):
  /data/data/com.termux/files/usr/share/cmake-3.20/Modules/CMakeTestCCompiler.cmake:26 (CMAKE_DETERMINE_COMPILER_ABI)
  CMakeLists.txt:2 (project)


-- Configuring incomplete, errors occurred!
See also "/data/data/com.termux/files/home/hello-world/bin/CMakeFiles/CMakeOutput.log".

```

--- a/packages/cmake/Modules-Platform-Android-GNU.cmake.patch
+++ b/packages/cmake/Modules-Platform-Android-GNU.cmake.patch
@@ -1,0 +1,17 @@
+--- a/Modules/Platform/Android-GNU.cmake
++++ b/Modules/Platform/Android-GNU.cmake
+@@ -24,6 +24,14 @@
+   return()
+ endif()
+ 
++# Natively compiling on an Android host doesn't use the NDK cross-compilation
++# tools.
++if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Android")
++  macro(__android_compiler_gnu lang)
++  endmacro()
++  return()
++endif()
++
+ include(Platform/Android-Common)
+ 
+ include(Platform/Android/abi-${CMAKE_ANDROID_ARCH_ABI}-GNU)

--- a/packages/cmake/build.sh
+++ b/packages/cmake/build.sh
@@ -4,6 +4,7 @@ TERMUX_PKG_LICENSE="BSD 3-Clause"
 TERMUX_PKG_MAINTAINER="@termux"
 # When updating version here, please update termux_setup_cmake.sh as well.
 TERMUX_PKG_VERSION=3.25.1
+TERMUX_PKG_REVISION=1
 TERMUX_PKG_SRCURL=https://www.cmake.org/files/v${TERMUX_PKG_VERSION:0:4}/cmake-${TERMUX_PKG_VERSION}.tar.gz
 TERMUX_PKG_SHA256=1c511d09516af493694ed9baf13c55947a36389674d657a2d5e0ccedc6b291d8
 TERMUX_PKG_DEPENDS="libarchive, libc++, libcurl, libexpat, jsoncpp, libuv, rhash, zlib"

